cspell.json

```diff
@@ -0,0 +1,173 @@
+{
+    "version": "0.2",
+    "words": [
+        // Spelling mistakes -- This includes using the english spelling
+        // of words over the american spelling.
+        "analyse",
+        "monomorphise",
+        "monomorphisation",
+        "optimisations",
+        "serialise",
+        "usefull",
+        "perfom",
+        "optimise",
+        "optimisation",
+        "occurences",
+        "puposes",
+        "flattenning",
+        "adress",
+        "lenghts",
+        "conresponding",
+        "initialised",
+        "occured",
+        "optimised",
+        "initialise",
+        "contrain",
```

Comment:
Is this the list of allowed misspellings? I don't think we should allow "contrain" or "concatened" for example.